tests/integration/components/bs-modal-test.js

```diff
@@ -312,5 +312,35 @@ module('Integration | Component | bs-modal', function (hooks) {
       await settled();
       assert.dom('.modal').doesNotExist();
     });
+
+    test('it allows to render modal in fullscreen using @fullscreen argument', async function (assert) {
+      this.set('fullscreen', null);
+      await render(hbs`<BsModal>`);
+      assert.dom('.modal-dialog').doesNotHaveClass(/modal-fullscreen/);
+
+      this.set('fullscreen', true);
+      await render(hbs`<BsModal @fullscreen={{this.fullscreen}}>`);
```

Comment:
No need to re-render again. The updated `{{this.fullscreen}}` value should be picked up automatically.

```suggestion
```

If it is not working, you may need to `await settled()` before to ensure Ember had time to pick up the state change and update the DOM. To be honest I'm still confused when `this.set()` in tests is triggering a sync re-render and when not.

It's the same for the other cases of course.